lib/event_store_client/adapters/grpc/commands/streams/read_paginated.rb

```diff
@@ -0,0 +1,141 @@
+                    options: options,
+                    skip_deserialization: true,
+                    skip_decryption: true
+                  ) do |opts|
+                    yield opts if block_given?
+                    position ||= get_position(opts)
+                    direction ||= get_direction(opts)
+                    max_count ||= opts.count.to_i
+
+                    raise(
+                      RecordsLimitError,
+                      "Pagination requires :max_count option to be greater than or equal to 2. Current value is `#{max_count}'."
+                    ) if max_count < 2
+                    paginate_options(opts, position)
+                  end
+                if response.success?
+                  processed_response =
+                    EventStoreClient::GRPC::Shared::Streams::ProcessResponses.new.call(
+                      response.success,
+                      skip_deserialization,
+                      skip_decryption
+                    )
+                  y << processed_response if processed_response.success.any?
+                  raise StopIteration if end_reached?(response.success, max_count)
+
+                  position = calc_next_position(response.success, direction, stream_name)
+                  raise StopIteration if position.negative?
+                else
+                  y << response
+                  raise StopIteration
+                end
+              end
+            end
+          end
+
+          private
+
+          # @param options [EventStore::Client::Streams::ReadReq::Options]
+          # @param position [Integer, nil]
+          # @return [EventStore::Client::Streams::ReadReq::Options, nil]
+          def paginate_options(options, position)
+            return unless position
+            return paginate_all_options(options, position) if options.stream.nil?
+
+            paginate_regular_options(options, position)
+          end
+
+          # @param options [EventStore::Client::Streams::ReadReq::Options]
+          # @param position [Integer]
+          # @return [EventStore::Client::Streams::ReadReq::Options]
+          def paginate_all_options(options, position)
+            options.all.position = EventStore::Client::Streams::ReadReq::Options::Position.new(
+              commit_position: position
+            )
+            options
+          end
+
+          # @param options [EventStore::Client::Streams::ReadReq::Options]
+          # @param position [Integer]
+          # @return [EventStore::Client::Streams::ReadReq::Options]
+          def paginate_regular_options(options, position)
+            options.stream.revision = position
+            options
+          end
+
+          # @param options [EventStore::Client::Streams::ReadReq::Options]
+          # @return [Symbol] :Backwards or :Forwards
+          def get_direction(options)
+            return options.read_direction if options.read_direction
+
+            :Forwards
```

Comment:
I would put it to the const like `DEFAULT_DIRECTION = :Forwards` and use it in all places where you repeated it.

Reply:
Make sense